### PR TITLE
[ESIMD] Fix tests to reflect APIs changes.

### PR DESCRIPTION
**Pull request by kbobrovs**
Complementary patch for https://github.com/intel/llvm/pull/4800.

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

**Comment by kbobrovs** [comment]
@v-klochkov, please review

**Comment by kbobrovs** [comment]
Test failures are expected, because they require updated compiler (intel/llvm#4800)

[2021-10-25T06:59:35.130Z]   SYCL :: DeprecatedFeatures/ESIMD/histogram_256_slm_spec.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: DeprecatedFeatures/ESIMD/spec_const_float.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: DeprecatedFeatures/ESIMD/spec_const_int.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: DeprecatedFeatures/ESIMD/spec_const_redefine_esimd.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: DeprecatedFeatures/ESIMD/spec_const_short.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: DeprecatedFeatures/ESIMD/spec_const_uint.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: DeprecatedFeatures/ESIMD/spec_const_ushort.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/BitonicSortKv2.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/PrefixSum.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/Prefix_Local_sum1.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/Prefix_Local_sum2.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/Prefix_Local_sum3.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/accessor_gather_scatter.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/accessor_load_store.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/api/ballot.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/api/esimd_bit_ops.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/api/slm_gather_scatter.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/api/slm_gather_scatter_heavy.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/api/slm_gather_scatter_rgba.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/ext_math.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/histogram_256_slm.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/histogram_raw_send.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/kmeans/kmeans.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/regression/usm_gather_scatter_32.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/slm_barrier.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/slm_split_barrier.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/spec_const/spec_const_float.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/spec_const/spec_const_int.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/spec_const/spec_const_short.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/spec_const/spec_const_uint.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/spec_const/spec_const_ushort.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/usm_gather_scatter_rgba.cpp
[2021-10-25T06:59:35.130Z]   SYCL :: ESIMD/vadd_raw_send.cpp

File: SYCL/ESIMD/spec_const/Inputs/spec-const-2020-common.hpp [hunk 1]
```diff
@@ -50,13 +51,13 @@ int main(int argc, char **argv) {
...
-          do_store(acc, i, kh.get_specialization_constant<ConstID>());
-        });
-      });
+         auto acc = buf.get_access<sycl::access::mode::write>(cgh);
+         if (i % 2 != 0)
+           cgh.set_specialization_constant<ConstID>(REDEF_VAL);
+         cgh.single_task<TestKernel>([=](kernel_handler kh) SYCL_ESIMD_KERNEL {
+           do_store(acc, i, kh.get_specialization_constant<ConstID>());
+         });
+       }).wait();
```
Comment (v-klochkov) [inline SYCL/ESIMD/spec_const/Inputs/spec-const-2020-common.hpp:60]: Just a comment: If I remember it correctly, the operation must be completed before the buffer 'buf' can be destroyed. So, the call of wait() is not required here. But it is Ok to have it here.

**Review by v-klochkov — APPROVED** [review]

File: SYCL/ESIMD/spec_const/Inputs/spec-const-2020-common.hpp (hunk 1 above)
Comment (kbobrovs) [reply SYCL/ESIMD/spec_const/Inputs/spec-const-2020-common.hpp:60]: you are right, buf is inside try block. This is a remnant from investigating another problem with `cgh.set_specialization_constant<ConstID>((T)0);`. I will update in a follow-up patch not to hold compiler patch.

**Pull request merged**